userspace/libsinsp/container_engine/docker_common.cpp

```diff
@@ -41,20 +161,106 @@ void docker::parse_json_mounts(const Json::Value &mnt_obj, vector<sinsp_containe
 	}
 }
 
-void docker::set_query_image_info(bool query_image_info)
+void docker_async_source::set_query_image_info(bool query_image_info)
 {
 	m_query_image_info = query_image_info;
 }
 
-bool docker::parse_docker(sinsp_container_manager* manager, sinsp_container_info *container, sinsp_threadinfo* tinfo)
+bool docker::resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, bool query_os_for_missing_info)
+{
+	std::string container_id, container_name;
+	sinsp_container_info *existing_container_info;
+
+	if(!g_docker_info_source)
```

Comment:
Can this be moved below, after the `detect_docker` call? I understand this spawns the async lookup thread so we could avoid it when not running docker

Reply:
Sure.